packages/lib/src/index.ts

```diff
@@ -42,10 +42,16 @@ export function isMessageOutput(output: Output): output is MessageOutput {
 }
 
 // CompletionOptions allows you to specify options for the completion request.
-export interface CompletionOptions<TModel> {
+export interface CompletionOptions<
+  TModel,
+  TInput extends Input = MessagesInput
+> {
   // If specified, partial updates will be streamed to this handler as they become available,
   // and only the first partial update will be returned by the Promise.
-  onStreamResult?: (result: Output | null, error: string | null) => unknown
+  onStreamResult?: (
+    result: TInput extends MessagesInput ? MessageOutput : Output,
```

Comment:
shouldn't it be `TInput extends MessagesInput ? MessageOutput : TextOutput,`?

Reply:
I was thinking that there might be API that returns MessageOutput regardless. We can add more condition to this check too.